components/level/game.tsx

```diff
@@ -666,9 +667,17 @@ export default function Game({
   useEffect(() => {
     if (gameState.board[gameState.pos.y][gameState.pos.x].levelDataType === TileType.End &&
       gameState.moves.length <= level.leastMoves && onComplete) {
+      const enrichedLevel = level as EnrichedLevel;
+
+      if (enrichedLevel.userMoves === undefined || gameState.moves.length <= enrichedLevel.userMoves) {
+        if (enrichedLevel.userMoves === undefined || gameState.moves.length < enrichedLevel.userMoves) {
+          saveCheckpoint(USER_BEST_MOVE_CHECKPOINT_SLOT);
```

Comment:
It may be safer to do this within `api/stats` if possible (there could be outdated state in the UX if the user has multiple tabs open / network errors)